frontend/micro-ui/web/micro-ui-internals/packages/modules/Estimate/src/pages/employee/Estimates/CreateDetailedEstimate/createEstimatePayload.js

```diff
@@ -4,41 +4,68 @@ const convertNumberFields=(text="")=>{
     return parseFloat(text)==0?null:parseFloat(text);
 
 }
-const transformLineItems = (SorData = []) => {
+const transformLineItems = (SorData = [], category, isEdit = false) => {
   const lineItems = [];
   SorData?.map((row) => {
     const measures = row?.measures?.map((measure) => {
-      return transformMeasure(measure, row);
+      return transformMeasure(measure, row, isEdit, category);
     });
     lineItems.push(...measures);
   });
   return lineItems;
 };
 
-function transformMeasure(measure, parentData) {
-  return {
-    sorId: parentData?.category == "SOR" ? parentData?.sorId : parentData?.sNo,
-    category: parentData?.category,
-    name: parentData?.description,
-    unitRate: parentData?.unitRate,
-    noOfunit: convertNumberFields(measure?.noOfunit),
-    uom: parentData?.uom,
-    height: convertNumberFields(measure?.height),
-    isDeduction: measure?.isDeduction,
-    length: convertNumberFields(measure?.length),
-    quantity: convertNumberFields(measure?.number),
-    uomValue: null,
-    width: convertNumberFields(measure?.width),
-    description: measure.description,
-    additionalDetails: parentData.additionalInfo, // Include data from parent object
-    amountDetail: [
-      {
-        type: "EstimatedAmount",
-        amount: parentData?.unitRate * measure?.noOfunit,
-        additionalDetails: {},
-      },
-    ],
-  };
+function transformMeasure(measure, parentData, isEdit, category) {
+  if(isEdit)
+    return {
+      sorId: parentData?.sorType ? parentData?.sorCode : parentData?.sNo,
+      id: measure?.id,
+      category: category,
```

Comment:
in edit scenario some usecase category is not coming in the parentData, backtracked it but has added this just to be sure it will not get missed anywhere.